pkg/metrics/prometheus_test.go

```diff
@@ -16,212 +16,102 @@
-		}
-	})
 
+	err = collector.AddCounter(ctx, "test_counter", 1, "label1")
+	if err != nil {
+		t.Fatal(err)
+	}
 }
 
-func Test_prom_NewCounter(t *testing.T) {
-	type fields struct {
-		counters map[string]prometheus.Counter
-		summary  map[string]prometheus.Summary
+func TestRegisterHistogram(t *testing.T) {
+	ctx := WithMetrics(context.Background(), "guac_test")
+	collector := FromContext(ctx, "guac_test")
+
+	_, err := collector.RegisterHistogram(ctx, "test_histogram", "label1")
+	if err != nil {
+		t.Fatal(err)
 	}
-	type args struct {
-		name string
+
+	err = collector.ObserveHistogram(ctx, "test_histogram", 2.5, "label1")
+	if err != nil {
+		t.Fatal(err)
 	}
-	tests := []struct {
-		name    string
-		fields  fields
-		args    args
-		wantErr bool
-	}{
-		{
-			name: "default",
-			fields: fields{
-				counters: make(map[string]prometheus.Counter),
-				summary:  make(map[string]prometheus.Summary),
-			},
-			args:    args{name: "test"},
-			wantErr: false,
-		},
-		{
-			name: "default",
-			fields: fields{
-				counters: make(map[string]prometheus.Counter),
-				summary:  make(map[string]prometheus.Summary),
-			},
-			args:    args{name: "test"},
-			wantErr: true,
-		},
+}
+
+func TestRegisterGauge(t *testing.T) {
+	ctx := WithMetrics(context.Background(), "guac_test")
+	collector := FromContext(ctx, "guac_test")
+
+	_, err := collector.RegisterGauge(ctx, "test_gauge", "label1")
+	if err != nil {
+		t.Fatal(err)
 	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			stop := make(chan struct{})
-			p := &prom{
-				counters: tt.fields.counters,
-				summary:  tt.fields.summary,
-			}
-			// this is a hack to make the test pass when the counter already exists
-			if tt.wantErr {
-				p.counters[tt.args.name] = prometheus.NewCounter(prometheus.CounterOpts{})
-			}
-			if err := p.NewCounter(tt.args.name); (err != nil) != tt.wantErr {
-				t.Errorf("NewCounter() error = %v, wantErr %v", err, tt.wantErr)
-			}
-			if tt.wantErr {
-				return
-			}
-
-			// start a server to get the metrics
-			http.Handle(
-				"/summary",
-				promhttp.Handler(),
-			)
-			go func() {
-				err := http.ListenAndServe(":2113", nil)
-				if err != nil {
-					fmt.Println(err)
-				}
-				close(stop)
-			}()
-			//increment the counter
-			p.IncrementCounter(tt.args.name)
-			// wait for the server to start
-			time.Sleep(1 * time.Second)
-			// get the metrics
-			resp, err := http.Get("http://localhost:2113/summary")
-			if err != nil {
-				t.Errorf("error getting metrics: %v", err)
-			}
-			body, err := httputil.DumpResponse(resp, true)
-			if err != nil {
-				t.Errorf("error reading response: %v", err)
-			}
-			found := false
-			for _, line := range strings.Split(string(body), "\n") {
-				if strings.Contains(line, fmt.Sprintf("%s_total", tt.args.name)) {
-					found = true
-				}
-			}
-			if !found {
-				t.Errorf("counter not found: %v", tt.args.name)
-			}
-		})
+	err = collector.SetGauge(ctx, "test_gauge", 1.5, "label1")
```

Comment:
should this also check the value after setting it?

Reply:
👍